src/middlewares/authMiddleware.js

```
@@ -0,0 +1,25 @@
+import { ApiError } from '../exeptions/api.error.js';
+import { jwtService } from '../services/jwt.service.js';
+
+export const authMiddleware = (req, res, next) => {
+  const authorization = req.headers['authorization'];
+
+  if (!authorization || !authorization.startsWith('Bearer ')) {
+    return next(ApiError.unauthorized());
+  }
+
+  const token = authorization.split(' ')[1];
+
+  try {
+    const userData = jwtService.verify(token);
+
+    if (!userData) {
+      return next(ApiError.unauthorized());
+    }
+
+    req.user = userData;
+    next();
+  } catch (err) {
+    return next(ApiError.unauthorized());
+  }
```

The catch block swallows the verification error and returns unauthorized (good for not leaking details). For development debugging, consider logging the JWT verification error when NODE_ENV !== 'production' so problems with tokens (e.g. malformed, expired, wrong secret) are easier to diagnose server-side. Keep client responses unchanged (generic Unauthorized).